frontend/z3_axioms.py

```diff
@@ -87,9 +76,8 @@ def arithmetic(left, right, result, is_mod, types):
         - "Case #%i: %i" % (u, v)
     """
     axioms = [
-        And(types.subtype(left, types.num), types.subtype(right, types.num),
-            Implies(types.stronger_num(left, right), result == left),
-            Implies(types.stronger_num(right, left), result == right))
+        And(types.subtype(left, types.num), types.subtype(right, left), result == left),
+        And(types.subtype(left, types.num), types.subtype(left, right), result == right),
```

Comment:
Same here:  And(types.subtype(right, types.num), types.subtype(left, right), result == right)